tt_metal/common/test_tiles.hpp

```diff
@@ -128,27 +153,31 @@ inline std::uint32_t round_up_to_mul16(std::uint32_t val) { return ((val & 15) =
 
 inline std::uint32_t round_up_to_mul32(std::uint32_t val) { return ((val & 31) == 0) ? val : (val | 31)+1; }
 
+inline std::uint32_t round_up_to_tile(int val, int tile_val) { return (val + tile_val - 1) & ~(tile_val - 1); }
+
 // Converts a linear non-zero-padded row-major tensor to zero-padded-32 32-swizzled tilized row-major tensor
 template <typename T, template <typename...> typename BufferType>
-inline std::vector<T> tilize_nchw(const BufferType<T>& in_rowmajor, const std::vector<std::uint32_t>& shape) {
+inline std::vector<T> tilize_nchw(const BufferType<T>& in_rowmajor, const std::vector<std::uint32_t>& shape, const std::optional<std::vector<uint32_t>>& tile_shape = std::nullopt) {
     ZoneScoped;
     int H = shape[shape.size() - 2], W = shape[shape.size() - 1];
     auto batch_size = 1;
     for (int i = 0; i < shape.size() - 2; i++) {
         batch_size *= shape[i];
     }
     int input_volume = batch_size * H * W;
-    int OH = round_up_to_mul32(H);
-    int OW = round_up_to_mul32(W);
+    auto tile_H = tile_shape.has_value() ? tile_shape.value()[0] : tt::constants::TILE_HEIGHT;
+    auto tile_W = tile_shape.has_value() ? tile_shape.value()[1] : tt::constants::TILE_WIDTH;
+    int OH = round_up_to_tile(H, tile_H);
+    int OW = round_up_to_tile(W, tile_W);
     std::vector<T> tilized_result;
     tilized_result.resize(batch_size * OH * OW);
     std::fill(tilized_result.begin(), tilized_result.end(), 0);
     int out_index = 0;
     for (auto batch_index = 0; batch_index < batch_size; batch_index++) {
-        for (int hs32 = 0; hs32 < H; hs32 += 32) {
-            for (int ws32 = 0; ws32 < W; ws32 += 32) {
-                for (int h32 = 0; h32 < 32; h32++) {
-                    for (int w32 = 0; w32 < 32; w32++) {
+        for (int hs32 = 0; hs32 < H; hs32 += tile_H) {
```

Comment:
The loop variable names are confusing since we changed the hard coded 32. A similar comment was made above.